src/ProgressOnderwijsUtils/Win32/Bitlocker.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Windows.Sdk;
+
+namespace ProgressOnderwijsUtils.Win32
+{
+    public static class BitLocker
+    {
+        static void Check(HRESULT result)
+        {
+            if (result.Value != 0) {
+                throw new Exception("HRESULT!=0");
+            }
+        }
+
+        public static unsafe int BitlockerStatusOfDriveLetter(char driveLetter)
+        {
+            Check(PInvoke.SHCreateItemFromParsingName(
+                $"{driveLetter}:",
+                null,
+                Guid.Parse(typeof(IShellItem2).GetCustomAttribute<GuidAttribute>().AssertNotNull().Value),
+                out var ppv

```

Comment:
IShellItem2 inherits from IShellItem inherits from IUnknown, en die heeft een [`Release`](https://docs.microsoft.com/en-us/windows/win32/api/unknwn/nf-unknwn-iunknown-release) method, blijkbaar om te de-alloceren.

Ik zou deze method in bijv linqpad of een debugger proberen en dan in een tight loop eens kijken hoe het zit met de memory leaks.

Ik verwacht dat je `ppv->Release()` nog moet doen, aan het einde hiervan, maar ik ken deze apis niet goed genoeg om te weten wat je verder allemaal nog moet Releasen.

Reply:
Even wat googlen vind bijv. hier iemand die duidelijk poogt lekker veel te releasen, en alsnog een memory leak vind; maar iig lijkt me wel zo duidelijk dat dit (Releasen) de bedoeling is: https://stackoverflow.com/questions/65641797/memory-leak-when-using-ishellitem2-getstring